administration/src/application/components/SteppedSubForms.tsx

```diff
@@ -0,0 +1,147 @@
+        return (
+          <Step key={index}>
+            <ButtonBase onClick={() => tryGoTo(index)} style={{ marginLeft: '-8px' }} disabled={loading}>
+              <StepLabel style={{ cursor: 'pointer', padding: '8px' }}>{label}</StepLabel>
+            </ButtonBase>
+            <StepContent>
+              <SubForm
+                validate={validate}
+                index={index}
+                setActiveStep={setActiveStep}
+                loading={loading}
+                onSubmit={index === subForms.length - 1 ? onSubmit : undefined}>
+                {element}
+              </SubForm>
+            </StepContent>
+          </Step>
+        )
+      })}
+    </Stepper>
+  )
+}
+
+type FormContextType = {
+  showAllErrors: boolean
+  disableAllInputs: boolean
+}
+const initialFormContext: FormContextType = { showAllErrors: false, disableAllInputs: false }
+
+export const FormContext = React.createContext<FormContextType>(initialFormContext)
+
+const SubForm = ({
+  children,
+  validate,
+  index,
+  setActiveStep,
+  onSubmit,
+  loading,
+}: {
+  children: ReactNode
+  validate: () => ValidationResult<unknown>
+  index: number
+  setActiveStep: SetState<number>
+  onSubmit?: () => void
+  loading: boolean
+}) => {
+  const [formContext, setFormContxt] = useState<FormContextType>(initialFormContext)
+  const { enqueueSnackbar } = useSnackbar()
+  useEffect(() => setFormContxt(state => ({ ...state, disableAllInputs: loading })), [loading])
+
+  const handleOnSubmit = useCallback(
+    (e: React.FormEvent<HTMLFormElement>) => {
+      e.preventDefault()
+      if (validate().type === 'error') {
+        enqueueSnackbar('Ihre Eingaben sind ungültig oder nicht vollständig.', { variant: 'error' })
+        setFormContxt(state => ({ ...state, showAllErrors: true }))
+      } else if (onSubmit === undefined) {
+        setActiveStep(() => index + 1)
+      } else {
+        onSubmit()
+      }
+    },
+    [validate, onSubmit, setActiveStep, index, enqueueSnackbar]
+  )
+  return (
+    <FormContext.Provider value={formContext}>
+      <form onSubmit={handleOnSubmit}>
+        {children}
+        <Divider sx={{ margin: '16px' }} />
+        {index === 0 ? null : (
+          <Button onClick={() => setActiveStep(() => index - 1)} disabled={formContext.disableAllInputs}>
+            Zurück
+          </Button>
+        )}
+        {onSubmit === undefined ? (
+          <Button type='submit' variant='contained' disabled={formContext.disableAllInputs}>
+            Nächster Schritt
+          </Button>
+        ) : (
+          <Button
+            type='submit'
+            variant='contained'
+            disabled={formContext.disableAllInputs}
+            endIcon={loading ? <CircularProgress size={20} color='inherit' /> : <Send />}>
+            Antrag Abschicken
+          </Button>
+        )}
+      </form>
+    </FormContext.Provider>
+  )
+}
+
+export const useFormAsStep = <
```

Comment:
Maybe move to separate file.

Reply:
I think I disagree here. This is just a very small 5-line helper to instantiate SteppedSubForms. Moving it to a separate file would make it harder to follow